src/components/Header/HeaderUserMenu/HeaderUserMenu.js

```diff
@@ -1,69 +1,99 @@
+                <p className="user-plan--monthly-text">
+                  <span>{user.plan.planName}</span> |{' '}
+                  <strong>
+                    {user.plan.maxSubscribers} {user.plan.itemDescription}
+                  </strong>
+                </p>
+              ) : (
+                ''
+              )}
+              <p>
+                <strong>{user.plan.remainingCredits}</strong>
+                {user.plan.description}
               </p>
+            </div>
+            {user.clientManager &&
+            user.plan.buttonUrl &&
+            user.plan.pendingFreeUpgrade !== 'true' ? (
+              <a className="buy-plan" href={user.plan.buttonUrl}>
+                {user.plan.buttonText}
+              </a>
+            ) : (
+              ''
+            )}
+            {!user.clientManager &&
+            user.plan.buttonUrl &&
+            user.plan.pendingFreeUpgrade !== 'true' ? (
+              <button onClick={this.handleOpenBuyModal} className="user-plan">
+                {user.plan.buttonText}
+              </button>
+            ) : (
+              ''
+            )}
+            {!user.clientManager && !user.plan.buttonUrl ? (
+              <button onClick={this.handleOpenBuyModal} className="user-plan">
+                {user.plan.buttonText}
+              </button>
             ) : (
               ''
             )}
-            <p>
-              <strong>{user.plan.remainingCredits}</strong>
-              {user.plan.description}
-            </p>
           </div>
-          {user.clientManager && user.plan.buttonUrl && user.plan.pendingFreeUpgrade !== 'true' ? (
-            <a className="buy-plan" href={user.plan.buttonUrl}>
-              {user.plan.buttonText}
-            </a>
-          ) : (
-            ''
-          )}
-          {!user.clientManager && user.plan.buttonUrl && user.plan.pendingFreeUpgrade !== 'true' ? (
-            <button className="buy-plan">{user.plan.buttonText}</button>
-          ) : (
-            ''
-          )}
-          {!user.clientManager && !user.plan.buttonUrl ? (
-            <button className="buy-plan">{user.plan.buttonText}</button>
-          ) : (
-            ''
-          )}
+          <ul className="options-user">
+            {user.nav.map((item, index) => (
+              <li key={index}>
+                <a href={item.url}>{item.title}</a>
+              </li>
+            ))}
+          </ul>
         </div>
-        <ul className="options-user">
-          {user.nav.map((item, index) => (
-            <li key={index}>
-              <a href={item.url}>{item.title}</a>
-            </li>
-          ))}
-        </ul>
+        <Modal
+          className="modal"
+          isOpen={this.state.buyModalIsOpen}
+          handleClose={this.handleCloseBuyModal}
+        >
+          <UpgradePlanForm idUserType={user.plan.isSubscribers === 'true' ? 4 : 2} handleClose={this.handleCloseBuyModal} />
```

Comment:
In my opinion, the id is like an implementation detail that we should move to the lowest layers.

Also, these string 'true' values are also a problem, what do you think about having a mapper layer below json file?
```suggestion
          <UpgradePlanForm isSuscribersPlan={user.plan.isSubscribers} handleClose={this.handleCloseBuyModal} />
```

But it is required some more work in lower layers:

* Map and normalize JSON result
* Understand semantic of _isSuscribersPlan_ property in API calls

Reply:
we depend on the json result, it's the same use in doppler. I do not know if it's worth investing time in this today

Reply:
Estuve trabajando en algo relacionado con eso ayer, te paso el commit:

https://github.com/andresmoschini/doppler-webapp/commit/0293e0c186628836676908a57ac7ba01b2cdc9ea

Si querés, dejá la parte del mapeo del JSON afuera de tu PR y luego cuando yo rebasee lo mío sobre lo tuyo hago el refactoring.

Tal vez si podés modificar la semantica de la prop que se le pasa al UpgradePlanForm, que en lugar de ser el id del userType sea una descripción del plan, luego, que Id usar se podría daterminar al momento de hacer el request al backend.

BTW, estuve haciendo varias pruebas que te pueden interesar, traté de meter un cambio atómico en cada commit (aunque en algunos no pude). Podés verlos en la rama [more-refactoring](https://github.com/andresmoschini/doppler-webapp/commits/more-refactoring), son los commits posteriores a "Merge branch 'session-abstraction' into merge1 "

![image](https://user-images.githubusercontent.com/1157864/53830022-4c3d3200-3f60-11e9-9411-b3d4e36f420e.png)